test/metabase/sync/analyze/fingerprint_test.clj

```diff
@@ -324,3 +329,16 @@
           (is (<= @#'sync.fingerprint/max-refingerprint-field-count attempted))
           ;; but it is bounded.
           (is (< attempted (+ @#'sync.fingerprint/max-refingerprint-field-count 10))))))))
+
+(deftest abandon-failed-fingerprint-test
+  (mt/test-drivers (mt/normal-drivers)
+    (testing "fingerprinting stops after the first table if we can't connect to the db"
+      (let [tables-fingerprinted (atom 0)]
+        (with-redefs [sync.fingerprint/fingerprint-fields! (fn [_table _fields]
+                                                             (swap! tables-fingerprinted inc)
+                                                             (throw (CannotAcquireResourceException. "Unable to acquire resource")))
+                      sync.fingerprint/*refingerprint?* true]
+          (is (=? (assoc (sync.fingerprint/empty-stats-map 0)
+                         :throwable #(instance? CannotAcquireResourceException %))
+                  (sync.fingerprint/fingerprint-fields-for-db! (mt/db) (fn [_ _] (log/debug _)))))
```

Comment:
This feels a bit weird. Would prefer `(constantly nil)` or naming the `_` arg if logging is still helpful.